homeassistant/components/vacuum/xiaomi_miio.py

```diff
@@ -383,3 +397,31 @@ def update(self):
             _LOGGER.error("Got OSError while fetching the state: %s", exc)
         except DeviceException as exc:
             _LOGGER.warning("Got exception while fetching the state: %s", exc)
+
+    async def async_clean_zone(self,
+                               zone,
+                               repeats: int = 1,
+                               repeats_template: str = None):
+        """Clean selected area for the number of repeats indicated."""
+        if repeats_template is not None:
+            _LOGGER.debug("Using template repeats")
+            repeats_template.hass = self.hass
+            repeats = repeats_template.async_render()
+            try:
+                repeats = int(repeats)
+            except ValueError:
+                repeats = 1
+                _LOGGER.debug("Failed to cast template. Set default > 1")
+        repeats = max(1, min(3, repeats))
+        _LOGGER.debug("Zone to clean: %s repeats: %s", zone, repeats)
+        from miio import DeviceException
+        _LOGGER.debug("Original zone: %s", zone)
+        for _zone in zone:
+            _zone.append(repeats)
+        _LOGGER.debug("Zone with repeats: %s", zone)
+        try:
+            self._vacuum.zoned_clean(zone)
```

Comment:
This isn't async safe as far as I can see. It needs to be scheduled on the executor thread pool via `hass.async_add_executor_job`.

Reply:
maybe in next release. thanks

Reply:
This needs to be fixed.

Reply:
Fixed